addon-test-support/@ember/test-helpers/settled.ts

```diff
@@ -41,6 +43,23 @@ function pendingRequests() {
   return localRequestsPending + internalRequestsPending;
 }
 
+/**
+  @private
+  @returns {boolean} if there are any pending router transitions
+*/
+function hasPendingTransitions() {
+  const context = getContext();
+  const owner = context && context.owner;
+
+  if (!owner) {
+    return false;
+  }
+
+  const router = owner.lookup('router:main');
```

Comment:
I would like to avoid doing this if we don't need it. Specifically, when there hasn't been a `visit` call, we don't need to check for transitions...